internal/util/ssa/cache.go

```diff
@@ -99,16 +125,16 @@ func (r *ssaCache) Has(key string) bool {
 // once we found out that it would not produce a diff.
 // The identifier consists of: gvk, namespace, name and resourceVersion of the original object and a hash of the modified
 // object. This ensures that we re-run the request as soon as either original or modified changes.
-func ComputeRequestIdentifier(scheme *runtime.Scheme, original, modified client.Object) (string, error) {
+func ComputeRequestIdentifier(scheme *runtime.Scheme, original, modified client.Object) (id, kind string, err error) {
```

Comment:
not sure if this is the best way to reuse the fact that we're pulling out GVK when computing the request ID

Reply:
I think we can revert the change to this func and simply use the Kind we already have available at the call sites of `Has`.
* dryRunSSAPatch only deals with Unstructured which always have to have Kind set. So we can use `dryRunCtx.originalUnstructured.GetKind()`
*  in ssa.Patch we have the gvk in l.79